installutils/helminstall/internal/memory_backed_rest_client_getter.go

```diff
@@ -0,0 +1,90 @@
+package internal
+
+import (
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery"
+	diskcached "k8s.io/client-go/discovery/cached/disk"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var (
+	defaultCacheDir = filepath.Join(clientcmd.RecommendedConfigDir, "http-cache")
+
+	// overlyCautiousIllegalFileCharacters matches characters that *might* not be supported.  Windows is really restrictive, so this is really restrictive
+	overlyCautiousIllegalFileCharacters = regexp.MustCompile(`[^(\w/\.)]`)
+
+	removeProtocol = regexp.MustCompile("http[s]?:/{2}")
+)
+
+func NewInMemoryRESTClientGetter(rawConfig clientcmd.ClientConfig) genericclioptions.RESTClientGetter {
```

Comment:
right, so I guess we only need this for helm stuff, so it's fine to stash it away in `internal/` here?

Reply:
I was confused by this as well, the existing code seemed to hide the HelmClient away in `internal` but exposed default constructors through in "installutils/helminstall/helm_client.go", so I adopted the same pattern here. Happy to change it to something else though